### DISCUSSION TITLE
Breaking Change Ideas

### DISCUSSION THREAD

**Discussion by YaLTeR** [YaLTeR/niri#447] [General]
Eventually we'll do a 0.2 or a 1.0, so let's collect ideas for what breaking changes we want to consider.

1. Remove deprecated four-number color format, i.e. `active-color 255 0 0 0`
    * These are all representable in the new CSS-like syntax.
1. Change linear animation settings to be set as properties similarly to spring animation settings, i.e. `linear duration-ms=200 curve="ease-out-quad"`
1. Change `toggle-debug-tint` to `debug-toggle-tint`
    * I've been intentionally prefixing new actions (i.e. `debug-toggle-damage`), this one should be prefixed too for consistency
    * As per a note in the debug wiki we can breaking-change this without a release, but it's not a terribly big deal to wait
1. KDL 2 (if a reasonable parsing crate exists by then)
    * This may enable more config improvements/breaking changes, haven't looked too deep into this

**Comment by YaLTeR** [comment]
We have multiple terms for outputs:
1. Output, i.e. `output "eDP-1" {}`
2. Monitor, i.e. `focus-monitor-left`
3. Screen, i.e. `screenshot-screen`

The reasoning for each is:
1. Output is the Wayland term
2. Monitor is a more "user-friendly" term
3. Screen comes from whatever it was called in GNOME Shell

Do we want to unify these in some way? I think we should at least replace `screen` since that screenshot action is the only place where it's used.

**Reply by rustysec** [reply]
i feel like `output` is aligned with other compositors :man_shrugging:

**Comment by 0xDEAD-C0DE** [comment]
I agree with the outputs  term instead of using screen or monitor. Can we remove the "off" values and just keep option values consistent with true or false? 
What is `debug-toggle-*`  or `toggle-debug-*` doing? This was my first question. I had to check the WIKI to find out that these are Keybindings not debug options. `show-tint`, `show-damage`, `show-opaque-regions` will be more descriptive. Thank you for your hard work.

**Reply by YaLTeR** [reply]
> Can we remove the "off" values and just keep option values consistent with true or false?

Why? I think off works pretty well.

> What is debug-toggle-* or toggle-debug-* doing? This was my first question. I had to check the WIKI to find out that these are Keybindings not debug options.

It's ok for debug options to require extra documentation, they're mainly useful for development.

> show-tint, show-damage, show-opaque-regions will be more descriptive

They should keep the word "debug" to make it clear that they are explicitly debug features, which means they can change at any time.

**Reply by 0xDEAD-C0DE** [reply]
You do an amazing work documenting everything. I was just suggesting that change to keep everything consistent. For the debugging options, it is best to keep `toggle-debug-*`. Users will know that we are toggling something on or off with those keybindings.
